linode/user/framework_datasource.go

```diff
@@ -0,0 +1,106 @@
+package user
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/linode/helper"
+)
+
+func NewDataSource() datasource.DataSource {
+	return &DataSource{}
+}
+
+type DataSource struct {
+	client *linodego.Client
+}
+
+func (d *DataSource) Configure(
+	ctx context.Context,
+	req datasource.ConfigureRequest,
+	resp *datasource.ConfigureResponse,
+) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	meta := helper.GetDataSourceMeta(req, resp)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	d.client = meta.Client
+}
+
+func (d *DataSource) Metadata(
+	ctx context.Context,
+	req datasource.MetadataRequest,
+	resp *datasource.MetadataResponse,
+) {
+	resp.TypeName = "linode_user"
+}
+
+func (d *DataSource) Schema(
+	ctx context.Context,
+	req datasource.SchemaRequest,
+	resp *datasource.SchemaResponse,
+) {
+	resp.Schema = frameworkDatasourceSchema
+}
+
+func (d *DataSource) Read(
+	ctx context.Context,
+	req datasource.ReadRequest,
+	resp *datasource.ReadResponse,
+) {
+	client := d.client
+
+	var data DataSourceModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	users, err := client.ListUsers(ctx, nil)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error listing Users: %s", err.Error(),
+		)
+		return
+	}
+
+	var user linodego.User
+
+	for _, testUser := range users {
+		if testUser.Username == data.Username.ValueString() {
+			user = testUser
+			break
+		}
+	}
+
+	if user.Username == "" {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Linode User with username %s was not found", data.Username.ValueString()), "",
+		)
+		return
+	}
```

Comment:
Can we move this check to some where before the API call? I think we can improve user experience this way by getting a faster hint to the users when they configure the `username` incorrectly.

Reply:
I think there needs to be an API call to check if the username the user is using actually exists. `user` is not set until after the API call because it depends on the returned list of usernames.